dcs/action.py

```diff
@@ -1901,52 +1909,55 @@ def dict(self) -> Dict[str, Any]:
                  horz_alignment: PictureAction.HorzAlignment, vert_alignment: PictureAction.VertAlignment,
                  size: int, size_units: PictureAction.SizeUnits) -> None:
         super().__init__(predicate=self.predicate, file_res_key=file_res_key,
                          seconds=seconds, clearview=clearview,
                          start_delay=start_delay, horz_alignment=horz_alignment,
                          vert_alignment=vert_alignment, size=size, size_units=size_units)
-        self.country_id = country_id
-        self.params = [self.country_id, self.file_res_key, seconds, clearview,
+        self.country = country
+        self.params = [self.country.id, self.file_res_key, seconds, clearview,
                        start_delay, horz_alignment, vert_alignment, size, size_units]
 
     @classmethod
-    def create_from_dict(cls, d, mission) -> "PictureToCountry":
-        return cls(d["countrylist"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+    def create_from_dict(cls, d: Dict[str, Any], mission: Mission) -> PictureToCountry:
+        return cls(countries.get_by_id(d["countrylist"]), ResourceKey(d["file"]),
+                   d["seconds"], d["clearview"], d["start_delay"],
                    d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
 
     def dict(self) -> Dict[str, Any]:
         d = super().dict()
-        d["countrylist"] = self.country_id
+        d["countrylist"] = self.country.id
         return d
 
 
 class PictureToGroup(PictureAction):
     predicate = "a_out_picture_g"
 
-    def __init__(self, group_id: int, file_res_key: ResourceKey, seconds: int,
+    def __init__(self, group: Group, file_res_key: ResourceKey, seconds: int,
                  clearview: bool, start_delay: int,
                  horz_alignment: PictureAction.HorzAlignment, vert_alignment: PictureAction.VertAlignment,
                  size: int, size_units: PictureAction.SizeUnits) -> None:
         super().__init__(predicate=self.predicate, file_res_key=file_res_key,
                          seconds=seconds, clearview=clearview,
                          start_delay=start_delay, horz_alignment=horz_alignment,
                          vert_alignment=vert_alignment, size=size, size_units=size_units)
-        self.group_id = group_id
-        self.params = [self.group_id, self.file_res_key, seconds, clearview,
+        self.group = group
+        self.params = [self.group.id, self.file_res_key, seconds, clearview,
                        start_delay, horz_alignment, vert_alignment, size, size_units]
 
     @classmethod
-    def create_from_dict(cls, d, mission) -> "PictureToGroup":
-        return cls(d["group"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+    def create_from_dict(cls, d: Dict[str, Any], mission: Mission) -> PictureToGroup:
+        group = mission.find_group_by_id(d["group"])
+        assert group is not None
```

Comment:
As in the other PR, don't use `assert` for error handling. Raise an exception.

Reply:
OK.